pkg/kando/location.go

```diff
@@ -38,17 +47,86 @@ func newLocationCommand() *cobra.Command {
 	return cmd.Flag(pathFlagName).Value.String()
 }
 
+// validateCommandArgs makes sure that we are getting exactly
+// one of --profile or --repository-server flags
+func validateCommandArgs(cmd *cobra.Command) error {
+	profileFlag := cmd.Flags().Lookup(profileFlagName).Value.String()
+	repositoryServerFlag := cmd.Flags().Lookup(repositoryServerFlagName).Value.String()
+	if profileFlag != "" && repositoryServerFlag != "" {
+		return errors.New("Either --profile or --repository-server can be provided")
+	}
+	if profileFlag == "" && repositoryServerFlag == "" {
+		return errors.New("Please provide either --profile or --repository-server as per the datamover you want to use")
+	}
+	return nil
+}
+
+func dataMoverFromCMD(cmd *cobra.Command, flag string) (datamover.DataMover, error) {
+	kopiaSnapshot, outputName := getKopiaSnapshotAndOutputNameFlag(cmd, flag)
+	dataMoverType, err := dataMoverTypeFromCMD(cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	switch dataMoverType {
+	case DataMoverTypeProfile:
+		profileRef, err := unmarshalProfileFlag(cmd)
+		if err != nil {
+			return nil, err
+		}
+		return datamover.NewProfileDataMover(profileRef, outputName, kopiaSnapshot), nil
+	case DataMoverTypeRepositoryServer:
+		repositoryServerRef, err := unmarshalRepositoryServerFlag(cmd)
+		if err != nil {
+			return nil, err
+		}
+		return datamover.NewRepositoryServerDataMover(repositoryServerRef, outputName, kopiaSnapshot), nil
+	default:
+		return nil, errors.New("Please provide either --profile or --repository-server as per the datamover you want to use")
+	}
+}
+
 func unmarshalProfileFlag(cmd *cobra.Command) (*param.Profile, error) {
 	profileJSON := cmd.Flag(profileFlagName).Value.String()
 	p := &param.Profile{}
 	err := json.Unmarshal([]byte(profileJSON), p)
 	return p, errors.Wrap(err, "failed to unmarshal profile")
 }
+
+func unmarshalRepositoryServerFlag(cmd *cobra.Command) (*param.RepositoryServer, error) {
+	repositoryServerJSON := cmd.Flag(repositoryServerFlagName).Value.String()
+	rs := &param.RepositoryServer{}
+	err := json.Unmarshal([]byte(repositoryServerJSON), rs)
+	return rs, errors.Wrap(err, "failed to unmarshal kopia repository server CR")
+}
+
+func dataMoverTypeFromCMD(c *cobra.Command) (DataMoverType, error) {
+	profile := c.Flags().Lookup(profileFlagName).Value.String()
+	if profile != "" {
+		return DataMoverTypeProfile, nil
+	}
+	repositoryServer := c.Flags().Lookup(repositoryServerFlagName).Value.String()
+	if repositoryServer != "" {
+		return DataMoverTypeRepositoryServer, nil
+	}
+	return "", errors.New("Please provide either --profile or --repository-server as per the datamover you want to use")
+}
+
+func getKopiaSnapshotAndOutputNameFlag(c *cobra.Command, flag string) (string, string) {
+	switch flag {
+	case kopiaSnapshotFlagName:
+		return c.Flag(kopiaSnapshotFlagName).Value.String(), ""
+	case outputNameFlagName:
+		return "", c.Flag(outputNameFlagName).Value.String()
+	default:
+		return "", ""
+	}
```

Comment:
```suggestion
	return c.Flag(kopiaSnapshotFlagName).Value.String(), c.Flag(outputNameFlagName).Value.String()
```
What is the default value `c.Flag().Value` returns if the flag not present? If it returns "", we don't need switch cases.

Reply:
The issue is, `kopiaSnapshotFlagName` is set only for pull and delete and `outputNameFlagName` is set for push. So let's say if we're trying to fetch value of `kopiaSnapshotFlagName` while pushing the data it will throw error as it's not being added in cobra command line in case of push. That's why we're using switch here